src/main/java/org/apache/commons/io/input/SequenceReader.java

```diff
@@ -60,8 +65,26 @@ public SequenceReader(final Reader... readers) {
      */
     @Override
     public void close() throws IOException {
+        if (readersIterable == null) {
+            // already closed
+            return;
+        }
+
+        final List<IOException> ioExceptionList = new ArrayList<>();
+        for (Reader reader : readersIterable) {
+            try {
+                reader.close();
+            } catch (IOException e) {
+                ioExceptionList.add(e);
```

Comment:
I'm not sure about the semantics here, which are completely different from `SequenceInputStream`. For good or bad, I think it would be better for the closing logic of this class to match `SequenceInputStream`.